--TEST--
Tests errors.

--INPUT--
using("std.ToStr");

try {
    read(using("file.Open")("sdjfisdjfidsfjisd"));
} catch x {
    writeln(x);
}

try {
    writeln("no problem");
} catch y {
    writeln("nope, not any problem");
}

try {
    writeln("Now a problem should occur");
    throw("Problem!!");
    writeln("not ran");
} catch z {
    writeln("problem found: " + ToStr(z));
} finally {
    writeln("DONE");
}

try {
    writeln("try / catch without error");
} catch e {
    writeln("Error on a try catch block without error?!");
} finally {
    writeln("Done");
}

a = 10;

try {
    writeln("test");
} catch e {
    writeln("error");
} finally if a == 10 {
    writeln("done");
}

try {
    throw("error");
} catch e if a == 15 {
    writeln("catched the error");
} finally if a == 10 {
    writeln("finally");
}

try {
    throw("error 2");
} catch e if a == 10 {
    writeln("catched it!");
} finally {
    writeln("done");
}

writeln("hello");

--OUTPUT--
java.nio.file.NoSuchFileException: sdjfisdjfidsfjisd
no problem
Now a problem should occur
problem found: Problem!!
DONE
try / catch without error
Done
test
done
finally
catched it!
done
hello